staging/src/k8s.io/client-go/transport/cache.go

```diff
@@ -91,6 +94,12 @@ func (c *tlsTransportCache) get(config *Config) (http.RoundTripper, error) {
 			KeepAlive: 30 * time.Second,
 		}).DialContext
 	}
+
+	// If we use a cert callback, we need to handle certificate rotation properly
+	if config.HasCertCallback() || config.TLS.ReloadTLSFiles {
```

Comment:
I expected this to gate only on whether or not you wanted to reload the TLS files.  If you don't want to reload, even if you have a GetCert function, why would we be reloading?

Reply:
The reload behaviour is intended to work arbitrarily on any callback to ensure that we rotate connections when the result of that callback changes. This allows us to remove the rotation logic in the exec auth plugin, for example. There are two cases:
1. You provided an explicit GetCert callback
2. You provided files but asked for them to be reloaded